src/compute-client/src/plan/mod.rs

```diff
@@ -1072,110 +1073,104 @@ impl<T: timely::progress::Timestamp> Plan<T> {
-                    let mut in_keys = arrangements
-                        .get(&Id::Global(*id))
-                        .cloned()
-                        .unwrap_or_else(AvailableCollections::new_raw);
-
-                    let (_, permutation, thinning) =
-                        in_keys.arranged.iter().find(|(k, _, _)| k == key).unwrap();
+                let input_mapper = JoinInputMapper::new(inputs);
+
+                // Plan each of the join inputs independently.
+                // The `plans` get surfaced upwards, and the `input_keys` should
+                // be used as part of join planning / to validate the existing
+                // plans / to aid in indexed seeding of update streams.
+                let mut plans = Vec::new();
+                let mut input_keys = Vec::new();
+                let mut input_arities = Vec::new();
+                for input in inputs.iter() {
+                    let (plan, keys) = Plan::from_mir_inner(input, arrangements, debug_info)?;
+                    input_arities.push(input.arity());
+                    plans.push(plan);
+                    input_keys.push(keys);
+                }
 
-                    mfp.permute(permutation.clone(), thinning.len() + key.len());
-                    in_keys.arranged = vec![(key.clone(), permutation.clone(), thinning.clone())];
-                    (
-                        Plan::Get {
-                            id: mz_expr::Id::Global(*id),
-                            keys: in_keys,
-                            plan: GetPlan::Arrangement(key.clone(), Some(val.clone()), mfp.take()),
-                        },
-                        AvailableCollections::new_raw(),
-                    )
-                } else {
-                    let input_mapper = JoinInputMapper::new(inputs);
-
-                    // Plan each of the join inputs independently.
-                    // The `plans` get surfaced upwards, and the `input_keys` should
-                    // be used as part of join planning / to validate the existing
-                    // plans / to aid in indexed seeding of update streams.
-                    let mut plans = Vec::new();
-                    let mut input_keys = Vec::new();
-                    let mut input_arities = Vec::new();
-                    for input in inputs.iter() {
-                        let (plan, keys) = Plan::from_mir_inner(input, arrangements, debug_info)?;
-                        input_arities.push(input.arity());
-                        plans.push(plan);
-                        input_keys.push(keys);
+                // Extract temporal predicates as joins cannot currently absorb them.
+                let (plan, missing) = match implementation {
+                    PredicateIndex(_id, key, _val) => {
+                        let start: usize = 1;
+                        let order = vec![(0usize, (*key).clone())];
```

Comment:
nit: you can write `key.clone()` here because [the `clone()` method accepts `&self`](https://doc.rust-lang.org/std/clone/trait.Clone.html).

Reply:
Thx